File: catkin_ws/src/learning_topic/scripts/pose_subscriber.py
#!/usr/bin/env python
import rospy
from turtlesim.msg import Pose

def poseCallback(msg):
    rospy.loginfo("Turtle pose: x:%0.6f, y:%0.6f", msg.x, msg.y)

def pose_subscriber():
    # ROS node init
    rospy.init_node('pose_subscriber', anonymous=True)

    # create Subscriber, topic /turtle/pose, callback function poseCallback
    rospy.Subscriber("/turtle1/pose", Pose, poseCallback)

    # wait
    rospy.spin()

if __name__ == '__main__':
    pose_subscriber()

File: catkin_ws/src/learning_topic/src/pose_subscriber.cpp
#include <ros/ros.h>
#include "turtlesim/Pose.h"

// callback function
void poseCallback(const turtlesim::Pose::ConstPtr& msg)
{
	// print
	ROS_INFO("Turtle pose: x:%0.6f, y:%0.6f", msg->x, msg->y);
}

int main(int argc, char **argv)
{
	// init ROS node
	ros::init(argc, argv, "pose_subscriber");

	// create handler
	ros::NodeHandle n;

	// create Subscriber, topic /turtle1/pose, callback function poseCallback
	ros::Subscriber pose_sub = n.subscribe("/turtle1/pose", 10, poseCallback);

	// loop wait
	ros::spin();

	return 0;
}
